Mathlib/Logic/Basic.lean

```diff
@@ -820,21 +820,21 @@ theorem exists_apply_eq (a : α) (b : β) : ∃ f : α → β, f a = b := ⟨fun
 @[simp] theorem exists_or_eq_right' (y : α) (p : α → Prop) : ∃ x : α, p x ∨ y = x := ⟨y, .inr rfl⟩
 #align exists_or_eq_right' exists_or_eq_right'
 
-theorem forall_apply_eq_imp_iff {f : α → β} {p : β → Prop} :
+theorem forall_apply_eq_imp_iff' {f : α → β} {p : β → Prop} :
     (∀ a b, f a = b → p b) ↔ ∀ a, p (f a) := by simp
-#align forall_apply_eq_imp_iff forall_apply_eq_imp_iff
+#align forall_apply_eq_imp_iff forall_apply_eq_imp_iff'
 
-@[simp] theorem forall_apply_eq_imp_iff' {f : α → β} {p : β → Prop} :
+@[simp] theorem forall_apply_eq_imp_iff {f : α → β} {p : β → Prop} :
     (∀ b a, f a = b → p b) ↔ ∀ a, p (f a) := by simp [forall_swap]
-#align forall_apply_eq_imp_iff' forall_apply_eq_imp_iff'
+#align forall_apply_eq_imp_iff' forall_apply_eq_imp_iff
```

Comment:
Is there a reason these can't both be `simp`?

Reply:
I'd really prefer that PR reviews are not routinely a place for exploring further unrelated changes. It bogs everything down.

Reply:
I don't think this is unrelated. You're saying "let's rename the simp lemmas to be the unprimed ones (and upstream only those?)", and I'm asking "why not make the unprimed ones be simp (and upstream all four)"

Reply:
(clearly I should have expressed that motivation in my previous comment though, sorry!)

Reply:
I think my point remains. There are three things that could happen here:

1. The lemmas I need get upstreamed (must happen)
2. The primes get swapped (convenient, because we don't actually use the unprimed lemmas, and I'm happy to do it as it makes upstreaming easier without the primes)
3. We add some additional simp attributes.

I have no interest in 3. If someone wants 3, they can do it! Just because I want to change this file, we don't need to have a discussion about other possible changes nearby.

Adding `simp` is "related" in the sense that it would touch the same lines as I'm touching here, but that is just a recipe for never getting anything done.

Adding `simp` is not related at all to what I am trying to get done.

Reply:
Maybe I misunderstood; are you upstreaming all four lemmas or just two?

Reply:
My hesitation here is that without more context, the primes already look the right way around; the primed lemma is in a sense "weirder" than the unprimed one. I'm not sold on the motivation "the primed one should be the main one because simp uses it", and especially not without knowing why we have the assymmetry.

Reply:
My guess is that the other lemma is not simpNF because another simpler `simp` lemma already handles the case with the binders in the wrong order

Reply:
I have no interest in the lemmas that are currently unprimed, and no intention of upstreaming them.

As far as I'm concerned deleting them from Mathlib would be a fine option.

Reply:
I don't understand this discussion: the other lemma is proved by simp, so a simp attribute on it would be useless.